numpy/ma/core.py

```diff
@@ -245,39 +275,29 @@ def default_fill_value(obj):
     (1e+20+0j)
 
     """
-    if hasattr(obj, 'dtype'):
-        defval = _check_fill_value(None, obj.dtype)
-    elif isinstance(obj, np.dtype):
-        if obj.subdtype:
-            defval = default_filler.get(obj.subdtype[0].kind, '?')
-        elif obj.kind in 'Mm':
-            defval = default_filler.get(obj.str[1:], '?')
+    def _scalar_fill_value(dtype):
+        if dtype.kind in 'Mm':
+            return default_filler.get(dtype.str[1:], '?')
```

Comment:
Just for comment: I think it's bad form to use `dtype.str` because it seems to be documented to relate to the PEP3118 interface, for use specifically in serialization operations. 

However, I accept that it is used this way in other places in numpy and that this is a pre-existing line which is hard to change, so I'm not asking you to change it.